src/pages/VoterDetail/Transactions/index.tsx

```diff
@@ -0,0 +1,119 @@
+/** @jsxImportSource @emotion/react */
+import React, { useMemo } from 'react';
+import BigNumber from 'bignumber.js';
+import { uid } from 'react-uid';
+import { Paper, Typography } from '@mui/material';
+import { Icon, AnchorButton, Table, TableProps, Spinner } from 'components';
+import { useTranslation } from 'translation';
+import { generateBscScanUrl, convertWeiToTokens } from 'utilities';
+import { VoteDetailTransaction } from 'types';
+import { useStyles } from './styles';
+
+interface ITransactionsProps {
+  className?: string;
+  address: string;
+  voterTransactions: VoteDetailTransaction[] | undefined;
+}
+
+export const Transactions: React.FC<ITransactionsProps> = ({
+  className,
+  address,
+  voterTransactions = [],
+}) => {
+  const styles = useStyles();
+  const { t } = useTranslation();
+  const columns = useMemo(
+    () => [
+      { key: 'action', label: t('voterDetail.actions'), orderable: false, align: 'left' },
+      { key: 'age', label: t('voterDetail.age'), orderable: false, align: 'left' },
+      { key: 'amount', label: t('voterDetail.amount'), orderable: false, align: 'right' },
+    ],
+    [],
+  );
+
+  const rows: TableProps['data'] = voterTransactions.map(voterTxs => {
+    // Transfer
+    let action = <></>;
+    let amountWei = new BigNumber(0);
+    if (voterTxs.type === 'transfer') {
+      action =
+        voterTxs.to.toLowerCase() === address.toLowerCase() ? (
+          <>
+            <Icon name="arrowShaft" css={styles.received} />
+            {t('voterDetail.receivedXvs')}
+          </>
+        ) : (
+          <>
+            <Icon name="arrowShaft" css={styles.sent} />
+            {t('voterDetail.sentXvs')}
+          </>
+        );
+      ({ amountWei } = voterTxs);
+    }
+    // Vote
+    if (voterTxs.type === 'vote') {
+      action = voterTxs.support ? t('voterDetail.receivedVotes') : t('voterDetail.lostVotes');
+      amountWei = voterTxs.votesWei;
+    }
+
+    return [
+      {
+        key: 'action',
+        render: () => (
+          <Typography css={styles.action} variant="small2" color="textPrimary">
+            {action}
+          </Typography>
+        ),
+        value: uid(action),
+        align: 'left',
+      },
+      {
+        key: 'age',
+        render: () => t('voterDetail.readableAge', { age: voterTxs.blockTimestamp }),
+        value: uid(voterTxs.blockTimestamp),
```

Comment:
Same thing here, it feels like uid isn't necessary

Reply:
There is a better way to solve that though

Reply:
Even using `uid` I should still end up with duplicates if two items are given the exact same value.

Perhaps you could append another prop to it?